src/basic/glob-util.c

```diff
@@ -29,14 +29,18 @@
 #include "path-util.h"
 #include "strv.h"
 
+static void _closedir(void* v) {
+        (void) closedir(v);
+}
+
 int safe_glob(const char *path, int flags, glob_t *pglob) {
         int k;
 
         /* We want to set GLOB_ALTDIRFUNC ourselves, don't allow it to be set. */
         assert(!(flags & GLOB_ALTDIRFUNC));
 
         if (!pglob->gl_closedir)
-                pglob->gl_closedir = (void (*)(void *)) closedir;
+                pglob->gl_closedir = _closedir;
```

Comment:
hmm, so gcc is ignoring our explicit cast here? that's weird...

Reply:
Oops, I pasted the wrong warning in the commit message. I repushed with an updated description.

In fact gcc is not warning about the assignment, but about the cast itself:
```
warning: cast between incompatible function types from 'int (*)(DIR *)' {aka 'int (*)(struct __dirstream *)'} to 'void (*)(void *)' [-Wcast-function-type]
```